tool.c

```diff
@@ -221,9 +222,83 @@ compdb(int argc, char *argv[])
 	return 0;
 }
 
+static void
+targets_rule(int argc, char *argv[])
+{
+	struct edge *e = NULL;
+	for (e = alledges; e; e = e->allnext) {
+		if (argc == 2) {
+			if (e->nin == 0)
+				continue;
+
+			/* TODO: what does `gen` means in a node? */
```

Comment:
`gen` is the edge that generates the node (the build action which produces the node as an output).